app/services/true_layer/sample_transactions_csv_parser.rb

```diff
@@ -0,0 +1,54 @@
+require 'csv'
+module TrueLayer
+  class SampleTransactionsCsvParser
+    TRANSACTIONS_CSV = 'db/sample_data/bank_transactions.csv'.freeze
+
+    def self.call
+      new.call
+    end
+
+    def call
+      @transaction_ids = []
+      CSV.parse(csv_file).map do |row|
```

Comment:
If you use `CSV.read` you can pass in the file path directly without first converting to a string in `csv_file`:
```ruby
CSV.read(Rails.root.join(TRANSACTIONS_CSV))
```

Reply:
Also if you add a header row to the CSV:
```csv
date,description,amount_out,amount_in 
27/09/2018,Cash Bnkm Cooperative,£40.00,
28/09/2018,Tombola.co.uk betting betting,£10.00,
28/09/2018,McDonalds Kilburn,£8.48,
```
You can do this:
```ruby
CSV.foreach(Rails.root.join(TRANSACTIONS_CSV), headers: true) do |row|
  p row['description']
end
```
It makes the code here a little more verbose, but makes the data handling more flexible because:

- The columns no longer need to be in exactly the same order as you've defined
- There can be extra columns that are ignored.
- The data itself is more informative in that they now have headers that describe each column.

Reply:
The later being particular important to clarify which row is `amount_out` and which `amount_in`

Reply:
Thanks! I'll do that